### PR TITLE
Cors fix

### PR DESCRIPTION
**Pull request by patrick-werner**
Using cors.allowed.origin=* together with credentials is forbidden:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Server-Side_Access_Control
https://www.w3.org/TR/cors/

and results in a 403 response.
This PR adds support for allowCredentials, and a list of allowed origins.
The spring CORS filter will check if the origin is in this list an returns the appropriate url.

### CODE REVIEW

File: src/main/java/ca/uhn/fhir/jpa/starter/HapiProperties.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
     static final String TESTER_CONFIG_REFUSE_TO_FETCH_THIRD_PARTY_URLS = "tester.config.refuse_to_fetch_third_party_urls";
     static final String CORS_ENABLED = "cors.enabled";
     static final String CORS_ALLOWED_ORIGIN = "cors.allowed_origin";
+    static final String CORS_ALLOWED_CREDENTIALS = "hapi.properties";
```
Comment (jamesagnew) [inline src/main/java/ca/uhn/fhir/jpa/starter/HapiProperties.java]: I guess this should probably be `CORS_ALLOW_CREDENTIALS` (no 'ed') and I think the value is wrong :)

Comment (patrick-werner) [reply src/main/java/ca/uhn/fhir/jpa/starter/HapiProperties.java]: you are right. Stupid minor mistakes. Fixed it

**Comment by patrick-werner** [comment]
@jamesagnew would love to merge, but i still don't have contributor rights on this repo.

**Comment by jamesagnew** [comment]
@patrick-werner - Can you check the email account you have registered with GitHub? You should see an invite.

**Comment by patrick-werner** [comment]
i got the invite, but i don't have rights within this project.

**Comment by jamesagnew** [comment]
Can you try again? I'm still learning how permissions work in GitHub after all these years apparently :p

**Comment by patrick-werner** [comment]
it worked. Thanks @jamesagnew

**Pull request merged**